app/models/reminder.rb

```diff
@@ -0,0 +1,47 @@
+class Reminder < ApplicationRecord
+  belongs_to :assignment
+  belongs_to :volunteer
+
+  default_scope { order(created_at: :desc) }
+
+  def self.create_for(volunteer, assignment)
+    Reminder.create! do |r|
+      r.volunteer = volunteer
+      r.assignment = assignment
+    end
+  end
+
+  def self.conditionally_create_reminders(time = Time.zone.now)
+    logger.info("Beginning reminder run, reference Date #{time}")
+    logger.flush
+
+    reminders_created_count = 0
+    Volunteer.all.find_each.count do |volunteer|
```

Comment:
Hmm should this be just `Volunteer.find_each do ...`? The `all` is definitely unnecessary here, and `count` just returns a number and doesn't take a block AFAIK?

Reply:
Also `.includes(:assigments)` should save some queries.